auth/httptransport/httptransport.go

```diff
@@ -129,18 +129,18 @@ type InternalOptions struct {
 
 // AddAuthorizationMiddleware adds a middleware to the provided client's
 // transport that sets the Authorization header with the value produced by the
-// provided [cloud.google.com/go/auth.TokenProvider]. An error is returned only
-// if client or tp is nil.
-func AddAuthorizationMiddleware(client *http.Client, tp auth.TokenProvider) error {
-	if client == nil || tp == nil {
+// provided [cloud.google.com/go/auth.Credentials]. An error is returned only
+// if client or creds is nil.
+func AddAuthorizationMiddleware(client *http.Client, creds *auth.Credentials) error {
+	if client == nil || creds == nil {
 		return fmt.Errorf("httptransport: client and tp must not be nil")
 	}
 	base := client.Transport
 	if base == nil {
 		base = http.DefaultTransport.(*http.Transport).Clone()
 	}
 	client.Transport = &authTransport{
-		provider: auth.NewCachedTokenProvider(tp, nil),
+		provider: creds,
 		base:     base,
 	}
```

Comment:
Should we call:

```
creds.TokenProvider = auth.NewCachedTokenProvider(creds.TokenProvider, nil)
```
here just in case the user did not construct the provided credentials with a wrapped token provider? The func is a no-op if the TokenProvider is already wrapped.

Reply:
OK, I understand this reasoning.